File: 2021/02.py
cs = [l.split(" ") for l in open("inputs/02.in").readlines()]

h,a,d = 0,0,0
for c,x in cs:
    if c == "forward":
        h += int(x)
        d += a * int(x)
    if c == "down":
        a += int(x)
    if c == "up":
        a -= int(x)

print(f"Part 1: {a*h}\nPart 2: {d*h}")

File: 2021/rust/02.rs
use std::string::String;

fn main() {
    let cs: Vec<(String, u32)> = std::fs::read_to_string("../inputs/02.in")
        .unwrap()
        .lines()
        .map(|l| {
            let (c, x) = l.split_once(" ").unwrap();
            (c.to_string(), x.parse::<u32>().unwrap())
        })
        .collect();

    let (mut h, mut d, mut a) = (0, 0, 0);
    for (c, x) in cs {
        match &c as &str {
            "forward" => {
                h += x;
                d += a * x;
            }
            "down" => a += x,
            "up" => a -= x,
            _ => {}
        }
    }

    println!("Part 1: {}\nPart 2: {}", a * h, d * h);
}
